# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1362139965 18000
#      Fri Mar 01 07:12:45 2013 -0500
# Node ID cb80b1d062b1f448884f5aedd0127951c98b2f0c
# Parent  fbc8f7afb1931fd92d903ea6af8c99a3d8d232cc
don't attempt to print unprintable tokens

* lex.ll (octave_lexer::display_token): Don't cast token values less
than 32 to char for printing.

diff --git a/libinterp/parse-tree/lex.ll b/libinterp/parse-tree/lex.ll
--- a/libinterp/parse-tree/lex.ll
+++ b/libinterp/parse-tree/lex.ll
@@ -3653,17 +3653,17 @@ octave_lexer::display_token (int tok)
     case METHODS: std::cerr << "METHODS\n"; break;
     case EVENTS: std::cerr << "EVENTS\n"; break;
     case CLASSDEF: std::cerr << "CLASSDEF\n"; break;
     case '\n': std::cerr << "\\n\n"; break;
     case '\r': std::cerr << "\\r\n"; break;
     case '\t': std::cerr << "TAB\n"; break;
     default:
       {
-        if (tok < 256)
+        if (tok < 256 && tok > 31)
           std::cerr << static_cast<char> (tok) << "\n";
         else
           std::cerr << "UNKNOWN(" << tok << ")\n";
       }
       break;
     }
 }
 
